apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -28,230 +24,95 @@ class TileDBObject(ABC, somacore.SOMAObject):
         Current open mode: read (r), write (w), or closed (None).
         """
-        return self._open_mode
-
-    @property
-    def closed(self) -> bool:
-        """
-        True iff self.mode is None
-        """
-        return self.mode is None
+        return self._handle.mode
 
-    @contextmanager
-    def _ensure_open(self, mode: OpenMode = "r") -> Iterator[None]:
-        """
-        Internal helper context for read/write methods to open self just for the duration of the
-        operation -- IFF self isn't already open. Also rejects attempts to write when self is
-        already open read-only.
-
-        when mode == "w":
-            if self is already open in read-only mode, raise an error.
-            if self is already open in w mode, do nothing.
-            otherwise open("w") until context exit.
-        when mode == "r":
-            if self is already open (IN EITHER MODE*), do nothing.
-            otherwise open("r") until context exit.
-
-            * We don't necessarily reject reads while we're open for writing, and different
-              timestamps may be used in this case. Usually, reads while open for writing just
-              pertain to metadata (e.g. exists()). The underlying TileDB object may reject other
-              cases.
-        """
-        assert mode in ("r", "w")
-        if self._open_mode:
-            if self._open_mode == "r" and mode == "w":
-                raise RuntimeError(
-                    self.__class__.__name__ + " is already open read-only"
-                )
-            yield
-        else:
-            with self.open_legacy(mode):
-                yield
-
-    @abstractproperty
-    def _tiledb_obj(self) -> Union[tiledb.Array, tiledb.Group]:
-        """
-        Get reference to open TileDB object handle (self must be open)
-        """
-        ...
-
-    def _common_create(self, soma_type: str) -> None:
-        """
-        This helps nested-structure traversals (especially those that start at the
-        Collection level) confidently navigate with a minimum of introspection on group
-        contents.
-        """
-        with self._ensure_open("w"):
-            self._tiledb_obj.meta.update(
-                {
-                    util.SOMA_OBJECT_TYPE_METADATA_KEY: soma_type,
-                    util.SOMA_ENCODING_VERSION_METADATA_KEY: util.SOMA_ENCODING_VERSION,
-                }
-            )
+    @classmethod
+    def _set_create_metadata(cls, handle: ReadWriteHandle[TDBHandle]) -> None:
+        """Sets the necessary metadata on a newly-created TileDB object."""
+        handle.writer.meta.update(
+            {
+                constants.SOMA_OBJECT_TYPE_METADATA_KEY: cls.soma_type,
+                constants.SOMA_ENCODING_VERSION_METADATA_KEY: constants.SOMA_ENCODING_VERSION,
+            }
+        )
+        # HACK: We need this so that the metadata appears on the read handle.
+        handle._flush_hack()
+
+    def _ensure_open_read(self) -> None:
```

Comment:
Nit: "ensure" might suggest it's going to open it if not already, maybe this is just a "check"